src/contexts/ModalContext.tsx

```diff
@@ -4,19 +4,33 @@ import { PropsWithChildren, createContext, useContext, useState } from 'react'
 
 export type ModalContextType = {
   isModalVisible: boolean
-  showModal: (state: boolean) => void
+  showModal: (element: React.ReactNode) => void
+  modalContent: React.ReactNode | null
+  closeModal: () => void
 }
 
 const ModalContext = createContext<ModalContextType | undefined>(undefined)
 
 export function ModalProvider({ children }: PropsWithChildren) {
-  const [isModalVisible, showModal] = useState(false)
+  const [isModalVisible, setIsModalVisible] = useState(false)
```

Comment:
**nitpick:** I think we don't need it now, as we can check it by checking if `modalContent` is not null